backend/mongodb_handler.py

```diff
@@ -0,0 +1,249 @@
+                    "approved": [],
+                    "content": {},
+                }
+            ).inserted_id
+
+        return None
+
+    def update_document(self, document_id: str, field: str, content: Any) -> None:
+        self.db.documents.update_one(
+            {"_id": ObjectId(document_id)}, {"$set": {field: content}}, upsert=False
+        )
+
+    def is_approved_by_company(self, document_id: str, company_name: str) -> bool:
+        document: Dict = cast(Dict, self.find_document(document_id))
+        lawyer_approve, economist_approve = False, False
+
+        for approved_by in document["approved"]:
+            user = self.find_user_by_id(ObjectId(approved_by))
+
+            if user is not None and user["company"] == company_name:
+                if user["role"] == Role.LAWYER:
+                    lawyer_approve = True
+                elif user["role"] == Role.ECONOMIST:
+                    economist_approve = True
+
+                if lawyer_approve and economist_approve:
+                    return True
+
+        return False
+
+    def delete_document(self, document_id: str, user_id: str) -> bool:
+        user: Dict = cast(Dict, self.find_user_by_id(ObjectId(user_id)))
+        document: Dict = cast(Dict, self.find_document(document_id))
+
+        if user["username"] == document["creator"]:
+            self.db.documents.delete_one({"_id": ObjectId(document_id)})
+            return True
+
+        return False
+
+    def document_exist(self, document_name: str) -> bool:
+        return (
+            self.db.documents.count_documents({"document_name": document_name}, limit=1)
+            != 0
+        )
+
+    def is_company_user_limit(self, company: str, role: str) -> bool:
+        user_limit = {
+            Role.LAWYER: 3,
+            Role.ECONOMIST: 3,
+            Role.GENERAL_DIRECTOR: 1,
+        }
+        return (
+            self.db.users.count_documents({"company": company, "role": role})
+            < user_limit[role]
+        )
+
+    def find_document(self, document_id: str) -> Optional[Dict]:
+        if ObjectId.is_valid(document_id):
+            return self.db.documents.find_one({"_id": ObjectId(document_id)})
+
+        return None
+
+    def select_document(self, company: str, user_id: ObjectId) -> List:
+        documents = self.select_company_documents(company)
+        permissions = self.get_user_permissions(user_id)
+
+        if len(permissions) != 0:
+            documents.extend(
+                [
+                    self.find_document(str(permission["document"]))
+                    for permission in permissions
+                ]
+            )
+
+        return documents
+
+    def select_company_documents(self, company: str) -> List:
+        return list(self.db.documents.find({"company": company}))
+
+    def select_company_users(self, company: str) -> List:
+        return list(self.db.users.find({"company": company}))
+
+    def leave_comment(
```

Comment:
`create_...`, `read_...`, `update_...`, `delete_...` -- common way to name methods in class/module